eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala

```diff
@@ -135,30 +136,31 @@ object PaymentRequest {
             expirySeconds: Option[Long] = None,
             extraHops: List[List[ExtraHop]] = Nil,
             timestamp: Long = System.currentTimeMillis() / 1000L,
-            features: Option[PaymentRequestFeatures] = Some(PaymentRequestFeatures(Features.VariableLengthOnion.mandatory, Features.PaymentSecret.mandatory))): PaymentRequest = {
-
+            paymentSecret: ByteVector32 = randomBytes32(),
+            features: PaymentRequestFeatures = PaymentRequestFeatures(Features.VariableLengthOnion.mandatory, Features.PaymentSecret.mandatory)): PaymentRequest = {
+    require(features.requirePaymentSecret, "invoices must require a payment secret")
```

Comment:
I don't think so, because it would force all payment requests to have this which would invalidate all the existing test vectors.
That's why I validated it only in that constructor:
* it forces our payment requests to set this
* but lets us deserialize payment requests that don't with `PaymentRequest.read`